src/schemas/small-catch.schema.js

```diff
@@ -13,22 +14,39 @@ export const createSmallCatchSchema = Joi.object({
     .when('noMonthRecorded', {
       is: true,
       then: Joi.required().messages({
-        'any.required': 'SMALL_CATCH_DEFAULT_MONTH_REQUIRED'
+        'any.required': 'SMALL_CATCH_DEFAULT_MONTH_REQUIRED',
+        'string.base': 'SMALL_CATCH_DEFAULT_MONTH_REQUIRED'
       }),
       otherwise: Joi.required().messages({
-        'any.required': 'SMALL_CATCH_MONTH_REQUIRED'
+        'any.required': 'SMALL_CATCH_MONTH_REQUIRED',
+        'string.base': 'SMALL_CATCH_MONTH_REQUIRED'
       })
     })
     .external(async (value, helper) => {
       const activityId = extractActivityId(helper.state.ancestors[0].activity)
 
+      // check duplicates
       const month = getMonthNumberFromName(value)
-
       const duplicateExists = await isDuplicateSmallCatch(activityId, month)
       if (duplicateExists) {
         return helper.message('SMALL_CATCH_DUPLICATE_FOUND')
       }
 
+      // check month in future
+      const submission = await getSubmissionByActivityId(activityId)
+
+      const currentDate = new Date()
+      const currentYear = getYear(currentDate)
+      const currentMonth = getMonth(currentDate) + 1 // getMonth is zero-based in date-fns
+
+      const inputMonth = getMonthNumberFromName(value) // Month is not one-based in our app
+      const inputDate = new Date(submission.season, inputMonth - 1) // Month is zero-based in js Date
+      const currentYearMonth = new Date(currentYear, currentMonth - 1) // Month is zero-based js Date
```

Comment:
The back-and-forth with the currentDate / currentYearMonth seems unnecessary. Rather than taking the current date, getting the year and month and then creating a new date, do this:
```
const currentYearAndMonth = set(new Date(), { date: 1, hours: 0, minutes: 0, seconds: 0, milliseconds: 0 })
```
`set` will obviously have to be imported from date-fns too